express-notes-api/index.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const app = express();
+const dataObj = require('./data.json');
+const port = 3000;
+const fs = require('fs');
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const key in dataObj.notes) {
+    arr.push(dataObj.notes[key]);
+  }
+  res.status(200).send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notesId = req.params.id;
+  if (notesId <= -1) {
+    res.status(400).send({ Error: 'the id you have entered is not positive' });
+  }
+  if (dataObj.notes[notesId]) {
+    res.status(200).send(dataObj.notes[notesId]);
+  }
+  if (!dataObj.notes[notesId]) {
```

Comment:
You could use an `else` here instead of writing `dataObj.notes[notesId]` multiple times. Also, it's usually best to take care of the error conditions first and then do the "happy path".